docs/git/index.md

```diff
@@ -0,0 +1,220 @@
+---
+id: index
+title: Introduction
+sidebar_label: Introduction
+---
+
+Most software engineers use a tool called Git to organise their code, and collaborate with other people.
+
+Git can be used for many things, and we will see lots of them throughout the course.
+
+Each week, you're going to use Git to get a copy of your homework exercises, to submit your solutions, and to get feedback on your solutions. Today, you're going to learn how to do these things.
+
+## Learning Objectives
+
+By the end of this class, you should be able to:
+
+* Get a copy of your homework questions onto your computer.
+  * (To _clone_ a repository).
+* Save your answers, and send them to our volunteers.
+  * (To _commit_ and _push_ changes).
+* See and respond to feedback on your homework.
+  * (To _fork_ a repository, use Github's _pull request_ workflows, and _push_ further changes).
+* Explore how a file has changed over time.
+  * (To use Git's _history log_).
```

Comment:
From @40thieves:

I'm a bit on the fence about the style of these LOs. I think ultimately there's an open question about who the audience for LOs is.

I historically have written them primarily aimed at teachers. Basically as tools for the teacher to assess whether they've done an effective job of teaching. However these can end up being a bit intimidating for students, so perhaps we should aim to write them more like this.

Reading my comment through above again, it's somewhat out-of-scope for this PR, so I would suggest that we discuss it another time.

---

Changing gears, what do you think about adding a LO along the lines of "Can summarise the reasons for using a VCS"?

As you'll see below, my overall feedback is that we're lacking context on why we're forcing students do all this arcane Git stuff.

Reply:
On the broader LOs question: I similarly suspect they're a bit intimidating - perhaps if they're for the teachers, we want to move them somewhere that isn't the top of the class page, given we point the learners at the syllabus... On the other hand, I quite like giving learner-oriented objects at the top of the syllabus, as it's maybe a handy index for when they're trying to recall something...

---

On the extra LO - I'm a bit torn. I suspect that if we _actually_ teach enough about a VCS for them to be able to summaries the reasoning well, we've probably overwhelmed them with a _lot_ of information and they probably won't have time to focus and grasp the key things we really _need_ of them from this week.

I tried to get away with this by framing the intro as "Git does a lot, we're going to learn a lot more, everyone uses it, here's your first lesson", rather than "Here's all the reasons this is valuable". I suspect if (as discussed below) we can make a bit more of checkpoints and discarding (in a very manual way, for now, and save things like revert/reset for the future), maybe we can get closer to this LO?

Reply:
I guess I forgot to post this comment before 😆 so here it is for posterity

> "Here's all the reasons this is valuable"

Right, good feedback. I was really thinking more along the lines of "Can summarise the reasons for using a VCS **for early stage students**", not all possible reasons. I.e. Git is good at things like sharing code between volunteers and students. Otherwise I worry that we fall into the trap of rote learning.